lib/coordinatesMapper.js

```diff
@@ -0,0 +1,29 @@
+// (c) Copyright 2021, SAP SE and ClearlyDefined contributors. Licensed under the MIT license.
+// SPDX-License-Identifier: MIT
+
+const defaultcache = require('../providers/caching/memory')({ defaultTtlSeconds: 60 * 60 * 24 /* 24 hours */ })
+const PypiCoordinatesMapper = require('./pypiCoordinatesMapper')
+
+const defaultMappers = {
+  pypi: new PypiCoordinatesMapper()
+}
+
+class CoordinatesMapper {
+
+  constructor(mappers = defaultMappers, cache = defaultcache) {
+    this.mappers = mappers
+    this.cache = cache
+  }
+
+  async map(coordinates) {
+    const mapper = this.mappers[coordinates?.provider]
```

Comment:
There's something going on with this line that's causing the `SyntaxError: Unexpected token '.'` (which I only found by running the mocha tests in debug mode in VS Code)

Reply:
I only see this error when parserOptions.ecmaVersion is changed to 9 in .eslintrc.json.  This should be addressed by my eslint upgrade commit.  Does eslint in build pipeline use config in formats other than json?